engine/src/main/scala/cromwell/server/CromwellShutdown.scala

```diff
@@ -0,0 +1,222 @@
+      *   engine job store) but still re-submitting all related metadata events. This is likely sub-optimal, but is used here
+      *   to simply stop the WMA (which will trigger all its descendants to be stopped recursively) without more coordination.
+      *   Indeed even if the actor is stopped in between the above mentioned gap, metadata will be re-submitted anyway on restart,
+      *   even for completed jobs.
+      *   
+      *  - Stop the LogCopyRouter: it can generate metadata events and must therefore be stopped before the service registry.
+      *   Wrap the ShutdownCommand in a Broadcast message so the router forwards it to all its routees
+      *   Use the ShutdownCommand because a PoisonPill could stop the routees in the middle of "transaction"
+      *   with the IoActor. The routees handle the ShutdownCommand properly and shutdown only when they have
+      *   no outstanding requests to the IoActor. When all routees are dead the router automatically stops itself.
+    */
+    shutdownActor(workflowStoreActor, CoordinatedShutdown.PhaseServiceRequestsDone, ShutdownCommand)
+    shutdownActor(logCopyRouter, CoordinatedShutdown.PhaseServiceRequestsDone, Broadcast(ShutdownCommand))
+    
+    /*
+      * Aborting is only a special case of shutdown. Instead of sending a PoisonPill, send a AbortAllWorkflowsCommand
+      * Also attach this task to a special shutdown phase allowing for a longer timeout.
+     */
+    if (abortJobsOnTerminate) {
+      val abortTimeout = coordinatedShutdown.timeout(PhaseAbortAllWorkflows)
+      shutdownActor(workflowManagerActor, PhaseAbortAllWorkflows, AbortAllWorkflowsCommand, Option(abortTimeout))
+    } else {
+      // This is a pretty rough shutdown of the WMA, depending on which route we go (full let it crash, or more fine grained
+      // state management) we might want to revisit this way of terminating the WMA and its descendants.
+      shutdownActor(workflowManagerActor, CoordinatedShutdown.PhaseServiceRequestsDone, PoisonPill)
+    }
+
+    /* 4) Shutdown connection pools
+      * This will close all akka http opened connection pools tied to the actor system.
+      * The pools stop accepting new work but are given a chance to execute the work submitted prior to the shutdown call.
+      * When this future returns, all outstanding connections to client will be terminated.
+      * Note that this also includes connection pools like the one used to lookup docker hashes.
+     */
+    coordinatedShutdown.addTask(CoordinatedShutdown.PhaseServiceStop, "TerminatingConnections") { () =>
+      Http(actorSystem).shutdownAllConnectionPools() as {
+        logger.info("Connection pools shut down")
+        Done
+      }
+    }
+
+    /* 5) Stop system level actors that require writing to the database or I/O
+      * - SubWorkflowStoreActor
+      * - JobStoreActor
+      * - CallCacheWriteActor
+      * - ServiceRegistryActor
+      * - DockerHashActor
+      * - IoActor
+    */
```

Comment:
Maybe drop a comment over where these are defined so that if someone adds a new one it gets added to the list